cmd/boulder-remoteva/main.go

```diff
@@ -0,0 +1,148 @@
+package notmain
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/letsencrypt/boulder/bdns"
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/features"
+	bgrpc "github.com/letsencrypt/boulder/grpc"
+	"github.com/letsencrypt/boulder/va"
+	vapb "github.com/letsencrypt/boulder/va/proto"
+)
+
+type Config struct {
+	RVA struct {
+		cmd.ServiceConfig
+
+		UserAgent string
+
+		IssuerDomain string
+
+		// DNSTries is the number of times to try a DNS query (that has a temporary error)
+		// before giving up. May be short-circuited by deadlines. A zero value
+		// will be turned into 1.
+		DNSTries    int
+		DNSProvider *cmd.DNSProvider `validate:"required_without=DNSStaticResolvers"`
```

Comment:
There are a bunch of copy-pasted fields here that we can extract into a common config object and then embed. It would look like this:

va/config/config.go:

```go
type VA struct {
  cmd.ServiceConfig
  UserAgent string
  IssuerDomain string
  DNSTries int
  ...
```

cmd/boulder-va/main.go:

```go
type Config struct {
  VA struct {
    config.VA // embeds the shared struct, meaning this effectively has all the same fields
    MaxRemoteValidationFailures int `validate:"omitempty,min=0,required_with=RemoteVAs"
    RemoteVAs                   []cmd.GRPCClientConfig `validate:"omitempty,dive"`
  }
}
```

cmd/boulder-rva/main.go:

```go
type Config struct {
  RVA config.VA // since the RVA's config fields are exactly equal to config.VA's fields we just use that type here, though we could also using struct embedding if we felt like it
}
```